src/session.coffee

```diff
@@ -226,32 +226,44 @@ class Session
   ###
   registerClick: (dfid, query) ->
     @set "dfid", dfid
-    @set "query", query  # not sure this is needed
-    @controller.registerClick dfid, sessionId: @get "session_id"
+    if query?
+      @set "query", query  # not sure this is needed
+    @client.registerClick dfid, 
+      sessionId: @get "session_id"
+      query: @get "query"
 
   ###*
    * Register a checkout when the location passed matches one of the URLs
    * provided.
    *
    * @public
-   * @param {String} location The URL to check against the patterns.
-   * @param {Array}  patterns A list of regular expressions to test with the
-   *                          provided location.
   ###
-  registerCheckout: (location, patterns = []) ->
+  registerCheckout: () ->
```

Comment:
Good, when `Session` was part of the layer it (almost) made sense, now it doesn't.